tools/json2bin/json2bin.py

```diff
@@ -134,22 +137,22 @@ def parse(fname_in: str, schema: Parser) -> bytes:
 def write(output_bin: bytes, output_idx: int, output_dir: str | None):
     output_fname = f'{output_idx:04}.bin'
     if output_dir:
-        output_fname = os.path.join(output_dir, output_fname)
+        output_fname = pathlib.Path(output_dir) / output_fname
     
-    with open(output_fname, 'wb') as output_file:
+    with open(output_fname, 'wb+') as output_file:
         output_file.write(output_bin)
 
 
 def process(fname_in: str,
             schema: Parser,
             output_dir: str | None,
             index_func: FunctionType):
-    if fname_in[-5:] != '.json':
+    if pathlib.Path(fname_in).suffix != '.json':
         return
     
     output_bin = parse(fname_in, schema)
-    simple_name = os.path.basename(os.path.normpath(fname_in))
-    output_idx = index_func(simple_name.split(".")[0].upper())
+    simple_name = pathlib.Path(fname_in).name
+    output_idx = index_func(pathlib.Path(simple_name).stem.upper())
```

Comment:
Apparently I misread this statement but rewrite it to be:
```python
    fname_inpath = pathlib.Path(fname_in)
    if fname_inpath .suffix != '.json':
        return

    output_bin = parse(fname_in, schema)
    simple_name = pathlib.Path(fname_in).name
    output_idx = index_func(fname_inpath.stem.upper())
```

Reply:
Sorry, it should be:
```python
```python
    fname_inpath = pathlib.Path(fname_in)
    if fname_inpath .suffix != '.json':
        return

    output_bin = parse(fname_in, schema)
    output_idx = index_func(fname_inpath.stem.upper())
```

Reply:
Yeah that's exactly what I ended up with.